src/properties.ts

```diff
@@ -33,15 +33,20 @@ import {
   SONAR_PROJECT_FILENAME,
 } from './constants';
 import { LogLevel, log } from './logging';
+import { getArch, getSupportedOS } from './platform';
 import { ScanOptions, ScannerProperties, ScannerProperty } from './types';
 
-const DEFAULT_PROPERTIES = {
-  [ScannerProperty.SonarUserHome]: path.join(
-    process.env.HOME ?? process.env.USERPROFILE ?? '',
-    SONAR_DIR_DEFAULT,
-  ),
-  [ScannerProperty.SonarScannerCliVersion]: SCANNER_CLI_VERSION,
-};
+function getDefaultProperties(): ScannerProperties {
+  return {
+    [ScannerProperty.SonarUserHome]: path.join(
+      process.env.HOME ?? process.env.USERPROFILE ?? '',
+      SONAR_DIR_DEFAULT,
+    ),
+    [ScannerProperty.SonarScannerCliVersion]: SCANNER_CLI_VERSION,
+    [ScannerProperty.SonarScannerOs]: getSupportedOS(),
+    [ScannerProperty.SonarScannerArch]: getArch(),
```

Comment:
I think I missed this, but we could add a test for checking the default parameters are set properly when nothing is provided

Reply:
Sounds good, we're already testing when nothing is provided that the default are used, but we're not testing that it can be overridden, I added a test for that!